### ISSUE TITLE
InvalidObjectState "The operation is not valid for the  current state of the object." during Warp IO run:

### ISSUE THREAD

**Issue by akmithal** [noobaa/noobaa-core#6680]
### Environment info
- NooBaa Version: master-20210721
- Platform: OCP 4.7

### Actual behavior
1. IO errors while doing warp run. Endpoint logs show error "The operation is not valid for the  current state of the object."

### Expected behavior
1. No error should come during warp run

### Steps to reproduce
1. Started multi user warp run for 2 hours

```
warp: Benchmark data written to "warp-mixed-2021-07-23[015631]-ei6j.csv.zst"
Mixed operations.
Operation: DELETE, 10%, Concurrency: 20, Duration: 1h59m56s.
 * Throughput: 1.41 obj/s

Operation: GET, 47%, Concurrency: 20, Duration: 2h0m0s.
Errors: 162
 * Throughput: 54.43 MiB/s, 6.34 obj/s

Operation: PUT, 12%, Concurrency: 20, Duration: 2h0m9s.
Errors: 3919
 * Throughput: 13.53 MiB/s, 1.58 obj/s

Operation: STAT, 31%, Concurrency: 20, Duration: 1h59m59s.
 * Throughput: 4.24 obj/s

Cluster Total: 67.96 MiB/s, 13.57 obj/s, 4081 errors over 2h0m0s.
Total Errors:4081.
warp: Cleanup Done.[root@hpo-11 ~]# warp mixed --access-key=nm21nfNliFIJx2PBWh9M --secret-key=3pOx5vX8DfYYuUwxbC8hbjPpJI9gGopCWCVZ9a4x --obj.size=9M --host=s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.-bucket=bucket-1 --debug --tls --insecure
```

This was seen on all instances of warp run.

-------
Jul-23 11:09:10.957 [Endpoint/9]    [L0] core.sdk.object_sdk:: validate_non_nsfs_bucket:  { name: SENSITIVE-7fa8f881394f8a6d, email: SENSITIVE-1979a16e9286fd8a, access_keys: [ { access_key: SENSIT
IVE-fbac2227580997b9, secret_key: SENSITIVE-633251558c397192 } ], has_login: false, has_s3_access: true, allowed_buckets: { full_permission: true }, default_resource: 'nsfs-nsr-1', can_create_buck
ets: true, nsfs_account_config: { gid: 800, uid: 5001, nsfs_only: false, new_buckets_path: '/', backend: 'GPFS', warn_threshold_ms: 100 }, systems: [ { name: 'noobaa', roles: [ 'admin' ] } ], exte
rnal_connections: { count: 0, connections: [] }, preferences: { ui_theme: 'DARK' } } { fs_root_path: '/nsfs/nsfs-nsr-1', fs_backend: 'GPFS', id: '60f9996c53a9980026b8c768', name: 'nsfs-nsr-1' }
Jul-23 11:09:10.957 [Endpoint/9]    [L1] core.endpoint.s3.s3_rest:: S3 REQUEST HEAD /bucket-2/8iPdj7QL/400.ei%29FzPeH57N3OrHO.rnd op head_object request_id krg8qsoc-12hdhr-dhh { 'user-agent': 'Min
IO (linux; amd64) minio-go/v7.0.11 warp/0.3.43', authorization: 'AWS4-HMAC-SHA256 Credential=eGs7YmGcf9TKPmOiI0Q7/20210723/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-
date, Signature=86cfb8d8e4eb427d71df9ffe540904162cf7ee4316f27b4b942274edd6e46b1b', 'x-amz-content-sha256': 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', 'x-amz-date': '202107
23T110910Z', host: 's3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com', 'x-forwarded-host': 's3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com', 'x-forwarded-port': '443', 'x-forwarded-proto': 'https', forwar
ded: 'for=10.17.66.193;host=s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com;proto=https', 'x-forwarded-for': '10.17.66.193' }
Jul-23 11:09:10.958 [Endpoint/9]    [L0] core.sdk.namespace_fs:: _fail_if_archived_or_sparse_file: /nsfs/nsfs-nsr-1/bucket-2/8iPdj7QL/501.dv8E7ySLd0quEdQ1.rnd rejected { dev: 1048592, ino: 1107976
76, mode: 33184, nlink: 1, uid: 5001, gid: 800, rdev: 0, size: 20000000, blksize: 4194304, blocks: 32768, atimeMs: 1627038544604.033, ctimeMs: 1627038546314.058, mtimeMs: 1627038546309.2761, birth
timeMs: 1627038546314.058, atime: 2021-07-23T11:09:04.604Z, mtime: 2021-07-23T11:09:06.309Z, ctime: 2021-07-23T11:09:06.314Z, birthtime: 2021-07-23T11:09:06.314Z }
Jul-23 11:09:10.959 [Endpoint/9] [ERROR] core.endpoint.s3.s3_rest:: S3 ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>InvalidObjectState</Code><**Message>The operation is not valid for the
 current state of the object.</Message>**<Resource>/bucket-2/8iPdj7QL/501.dv8E7ySLd0quEdQ1.rnd</Resource><RequestId>krg8qso3-x5b4v-sue</RequestId></Error> GET /bucket-2/8iPdj7QL/501.dv8E7ySLd0quEdQ1
.rnd {"user-agent":"MinIO (linux; amd64) minio-go/v7.0.11 warp/0.3.43","authorization":"AWS4-HMAC-SHA256 Credential=eGs7YmGcf9TKPmOiI0Q7/20210723/us-east-1/s3/aws4_request, SignedHeaders=host;x-am
z-content-sha256;x-amz-date, Signature=8829bd01d9f440a95d5c70730b21c8cbe92f3de1bd76e7871b21b75389ef3541","x-amz-content-sha256":"e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855","
x-amz-date":"20210723T110910Z","host":"s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com","x-forwarded-host":"s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com","x-forwarded-port":"443","x-forwarded-proto":"h
ttps","forwarded":"for=10.17.66.193;host=s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com;proto=https","x-forwarded-for":"10.17.66.193"} Error: Attempted to access archived or sparse file
    at NamespaceFS._fail_if_archived_or_sparse_file (/root/node_modules/noobaa-core/src/sdk/namespace_fs.js:1280:19)
    at async NamespaceFS.read_object_stream (/root/node_modules/noobaa-core/src/sdk/namespace_fs.js:477:13)
    at async ObjectSDK.read_object_stream (/root/node_modules/noobaa-core/src/sdk/object_sdk.js:396:23)
    at async Object.get_object [as handler] (/root/node_modules/noobaa-core/src/endpoint/s3/ops/s3_get_object.js:115:19)
    at async handle_request (/root/node_modules/noobaa-core/src/endpoint/s3/s3_rest.js:149:19)
    at async Object.s3_rest [as handler] (/root/node_modules/noobaa-core/src/endpoint/s3/s3_rest.js:68:9)
Jul-23 11:09:10.997 [Endpoint/9]    [L1] core.util.signature_utils:: _string_to_sign_v4 method PUT pathname /bucket-2/M7DizQqr/517.H4UX%29Oxy6tDO63Dp.rnd search  headers { 'User-Agent': 'MinIO (li
nux; amd64) minio-go/v7.0.11 warp/0.3.43', 'Content-Length': '20000000', Authorization: 'AWS4-HMAC-SHA256 Credential=eGs7YmGcf9TKPmOiI0Q7/20210723/us-east-1/s3/aws4_request, SignedHeaders=content-
type;host;x-amz-content-sha256;x-amz-date, Signature=67a38e0417a6b6bbf600c93c98cb10cbeb6589fa37eb7c0439551c7cf02f6bd7', 'Content-Type': 'application/octet-stream', 'X-Amz-Content-Sha256': 'UNSIGNE
D-PAYLOAD', 'X-Amz-Date': '20210723T110910Z', Host: 's3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com', 'X-Forwarded-Host': 's3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com', 'X-Forwarded-Port': '443', 'X-
Forwarded-Proto': 'https', Forwarded: 'for=10.17.66.193;host=s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com;proto=https', 'X-Forwarded-For': '10.17.66.193' } region us-east-1 canonical_str 
PUT
/bucket-2/M7DizQqr/517.H4UX%29Oxy6tDO63Dp.rnd

content-type:application/octet-stream
host:s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com
x-amz-content-sha256:UNSIGNED-PAYLOAD
x-amz-date:20210723T110910Z

content-type;host;x-amz-content-sha256;x-amz-date
UNSIGNED-PAYLOAD
 string_to_sign 
AWS4-HMAC-SHA256
20210723T110910Z

------

### More information - Screenshots / Logs / Other output

Uploading must-gather logs

**Comment by akmithal** [comment]
[must-gather-8.tar.gz](https://github.com/noobaa/noobaa-core/files/6868915/must-gather-8.tar.gz)

**Comment by jackyalbo** [comment]
@akmithal , It seems like we diagnosed this file as being sparsed or archived - and that's why reading it got blocked.
It uses 32768 blocks of 512bytes each ~ 16MB of storage size - which is less than the original 20MB file.
As I couldn't reproduce locally. How often do you see this issue? Do you have some sort of access to the file? Can you check that the original file and the data in the storage are the same (maybe with MD5)? Is it possible any compression is running on the data?

**Comment by guymguym** [comment]
Adding @eshelmarc

**Comment by eshelmarc** [comment]
@akmithal  do we have a gpfs trace? can you try to get it running just with the problematic file?

**Comment by davidbw246** [comment]
I see this same error(InvalidObjectState) on noobaa 0727 for all "aws s3 cp" commands doing GETs on all objects I have tested that are smaller than 4KB.
"aws s3 cp" commands doing GETs for objects >4KB work.
"aws s3 cp" doing PUTs works for all object sizes I have tested(sizes 328 bytes to 388MB).

**Comment by akmithal** [comment]
Hi @eshelmarc , I ran it again and it was easily reproducible. I have defined file sizes to be used for warp tool of 9MB. 
I don't have control to have this file separately as these get created randomly by tool.

Here is the error I get and I have collected gpfs.snap

```
Aug-4 9:02:23.827 [Endpoint/8] [ERROR] core.endpoint.s3.s3_rest:: S3 ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>InvalidObjectState</Code><Message>The operation is not valid for the c
urrent state of the object.</Message><Resource>/bucket-1/GjYhorxH/287.JeI%28vDuweZbmGxVr.rnd</Resource><RequestId>krx9hyw4-464aqo-x7i</RequestId></Error> GET /bucket-1/GjYhorxH/287.JeI%28vDuweZbmG
xVr.rnd {"user-agent":"MinIO (linux; amd64) minio-go/v7.0.11 warp/0.3.43","authorization":"AWS4-HMAC-SHA256 Credential=iCfVDDsVM1f0hx4WTizy/20210804/us-east-1/s3/aws4_request, SignedHeaders=host;x
-amz-content-sha256;x-amz-date, Signature=d71650bc3dc033cc71a1f6ea863b2c82f93f1107900674e142f5baf076fba25a","x-amz-content-sha256":"e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
","x-amz-date":"20210804T090223Z","host":"s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com","x-forwarded-host":"s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com","x-forwarded-port":"443","x-forwarded-proto"
:"https","forwarded":"for=10.17.66.193;host=s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com;proto=https","x-forwarded-for":"10.17.66.193"} Error: Attempted to access archived or sparse file
    at NamespaceFS._fail_if_archived_or_sparse_file (/root/node_modules/noobaa-core/src/sdk/namespace_fs.js:1280:19)
    at async NamespaceFS.read_object_stream (/root/node_modules/noobaa-core/src/sdk/namespace_fs.js:477:13)
    at async ObjectSDK.read_object_stream (/root/node_modules/noobaa-core/src/sdk/object_sdk.js:396:23)
    at async Object.get_object [as handler] (/root/node_modules/noobaa-core/src/endpoint/s3/ops/s3_get_object.js:115:19)
    at async handle_request (/root/node_modules/noobaa-core/src/endpoint/s3/s3_rest.js:149:19)
    at async Object.s3_rest [as handler] (/root/node_modules/noobaa-core/src/endpoint/s3/s3_rest.js:68:9)
```

**Comment by akmithal** [comment]
[must-gather-0408.tar.gz](https://github.com/noobaa/noobaa-core/files/6930110/must-gather-0408.tar.gz)

gpfs.snap logs are provided in slack as it is not getting uploaded here.

**Comment by nimrod-becker** [comment]
Should be fixed with #6698

**Comment by akmithal** [comment]
This defect is fixed with the recent fixes in build master-20210812 onwards. Closing it.
